### DISCUSSION TITLE
Heya! How's the lv2 goal coming along?

### DISCUSSION THREAD

**Discussion by TheGuyWhoo** [Wasted-Audio/hvcc#41] [General]
I super interested in that, would indeed be very cool!

**Comment by dromer** [comment]
Hey, sorry just read this. LV2 totally works already if you use the DPF generator.
It can also create dssi, ladspa, vst2, and soon vst3 using the [DISTRHO Plugin Framework](https://github.com/DISTRHO/DPF).
See [docs](https://wasted-audio.github.io/hvcc/docs/03.gen.dpf.html) and [examples](https://github.com/Wasted-Audio/hvcc/tree/develop/examples/dpf ) for more info on how to use this generator.

Some features still todo:
* MIDI_RT_CONTINUE
* LV2 cv-ports
* enumerable message type (list of strings that you could select in a UI)
* probably much more :)

See the [project page](https://github.com/Wasted-Audio/hvcc/projects/1) for more.

**Reply by TheGuyWhoo** [reply]
Nice! That's sweet. Just spent the morning testing it out and so far so good.

**Reply by dromer** [reply]
Good to hear! Am always interested in example plugins we can add (want to start a separate repo for each of the generators with this purpose soon)

**Reply by TheGuyWhoo** [reply]
It would be great to have something of mine added!

I'm having a bit of trouble though. In this patch I made a polyphonic supersaw synth. It compiles without issue, and in the daw works perfectly when playing it with a midi keyboard. But when I try to sequence it with the piano roll in reaper, there is a lot of latency.

Would you mind having a look?

[chaos_saw.zip](https://github.com/Wasted-Audio/hvcc/files/8541621/chaos_saw.zip)

(Just a note, G_or_L_detune stands for global or local detune. When it's on detuners can be controlled separately, but when it's off the first detuner controls them all)

**Reply by dromer** [reply]
Why not use a single `[route 0 1 2 3]` and then a abstraction for the synthvoice that you can duplicate 4 times?

It mostly seems to work .. but I get a lot of stuck notes :#

**Reply by TheGuyWhoo** [reply]
Hmm tried those suggestions and it didn't really make a difference. I can't figure this one out, very strange that it can be played in real time with a midi keyboard but has extreme lag from a midi file.

**Reply by dromer** [reply]
I found a bug with scheduling midi messages. Fixed in latest develop -> https://github.com/Wasted-Audio/hvcc/commit/9fe1ea149e8d1314e4cea01bb467e4f540f2bca0
These delayed messages should hopefully be resolved now :)

We also now have a '[stripnote]' abstraction that could help with more custom note off handling and such.

**Reply by TheGuyWhoo** [reply]
Sweet, I will take a look! Thank you.
